src/java/org/apache/cassandra/index/sai/disk/format/IndexComponent.java

```diff
@@ -73,7 +73,12 @@ public enum IndexComponent
     TOKEN_VALUES("TokenValues"),
 
     /**
-     * An on-disk block packed index containing the starting and ending rowIds for each partition.
+     * An on-disk block packed index mapping rowIds to partitionIds.
+     */
+    PARTITION_ROWS("PartitionRows"),
+
+    /**
+     * An on-disk block packed index containing the partition size for each partition keyed by partitionId.
```

Comment:
I would take some care to be clear everywhere in the new code that partition size is the number of rows.

Reply:
This was on my list to look at the naming and associated comments of these on-disk components. I have gone with renaming all of them to make it clearer exactly what they contain. I have also made sure the users of these (in the primary key maps) are correctly named and commented.

I have to admit I was getting confused by the naming and what they were meant to be doing.